### PR TITLE
Fixed #1697 ensuring that we properly rename the shadow state properties...

### LINKED ISSUE #1697: Foreign key name is incorrectly generated for one:many relationships
In the below model, we create a shadow state property named PersonId while we are building the Siblings relationship rather than a SiblingReverseId property
```
    public class Person
    {
        public int Id { get; set; }
        public string Name { get; set; }

        public List<Person> Siblings { get; set; }
        public Person Lover { get; set; }
        public Person LoverReverse { get; set; }
        public Person SiblingReverse { get; set; }
    }

    public class GoTContext : DbContext
    {
        public DbSet<Person> People { get; set; }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<Person>().HasMany(p => p.Siblings)
                .WithOne(p => p.SiblingReverse).Required(false);
            modelBuilder.Entity<Person>().HasOne(p => p.Lover)
                .WithOne(p => p.LoverReverse).Required(false);
        }

        protected override void OnConfiguring(DbContextOptions options)
        {
            options.UseSqlServer(@"Server=(localdb)\mssqllocaldb;Database=StateManagerBug;Trusted_Connection=True;MultipleActiveResultSets=true");
        }
    }
```

### PR DESCRIPTION
**Pull request by lukewaters**
... in

a foreign key relationship

### CODE REVIEW

File: test/EntityFramework.SqlServer.FunctionalTests/ComplexNavigationsQuerySqlServerTest.cs [hunk 1]
```diff
@@ -17,21 +19,21 @@ public ComplexNavigationsQuerySqlServerTest(ComplexNavigationsQuerySqlServerFixt
         public override void Multi_level_include_one_to_many_optional_and_one_to_many_optional_produces_valid_sql()
         {
             base.Multi_level_include_one_to_many_optional_and_one_to_many_optional_produces_valid_sql();
-
+            Debug.Write(Sql);
```
Comment (lajones) [inline test/EntityFramework.SqlServer.FunctionalTests/ComplexNavigationsQuerySqlServerTest.cs]: Did you mean to leave this in?

File: test/EntityFramework.SqlServer.FunctionalTests/NavigationTest.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+                        modelBuilder.Entity<Person>().HasOne(p => p.SiblingReverse).WithMany(p => p.Siblings).Required(false);
+                        modelBuilder.Entity<Person>().HasOne(p => p.Lover).WithOne(p => p.LoverReverse).Required(false);
+                        return 0;
+                    };
+
+                var model = context.Model;
+                var entityType = model.EntityTypes.First();
+
+                Console.WriteLine(entityType);
```
Comment (lajones) [inline test/EntityFramework.SqlServer.FunctionalTests/NavigationTest.cs]: Did you mean to leave this in?

File: test/EntityFramework.SqlServer.FunctionalTests/NavigationTest.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+                        modelBuilder.Entity<Person>().HasMany(p => p.Siblings).WithOne(p => p.SiblingReverse).Required(false);
+                        modelBuilder.Entity<Person>().HasOne(p => p.Lover).WithOne(p => p.LoverReverse).Required(false);
+                        return 0;
+                    };
+
+                var model = context.Model;
+                var entityType = model.EntityTypes.First();
+
+                Console.WriteLine(entityType);
```
Comment (lajones) [inline test/EntityFramework.SqlServer.FunctionalTests/NavigationTest.cs]: Looks like a debug thing - did you mean to leave this in?

Comment (lukewaters) [inline test/EntityFramework.SqlServer.FunctionalTests/NavigationTest.cs]: :+1: will remove

File: test/EntityFramework.SqlServer.FunctionalTests/NavigationTest.cs (hunk 2 above)
Comment (lukewaters) [inline test/EntityFramework.SqlServer.FunctionalTests/NavigationTest.cs]: :+1: will remove

File: test/EntityFramework.SqlServer.FunctionalTests/ComplexNavigationsQuerySqlServerTest.cs (hunk 1 above)
Comment (lukewaters) [inline test/EntityFramework.SqlServer.FunctionalTests/ComplexNavigationsQuerySqlServerTest.cs]: :+1: will remove

**Comment by lajones** [comment]
:shipit:

**Pull request merged**